File: Bachelor/Semester6-Finals/MPP - Projecting and Programing Methods/tReservation-Lab1CS/tReservation-Lab1CS/repository/ICourseRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using tReservation_Lab1CS;

namespace tReservation_Lab1CS
{
    interface ICourseRepository
    {
        int size();
        void save(Course course);
        void delete(int id);
        void update(int id, Course course);
        Course FindOne(int courseId);
        Course FindOne(string departureCity,string destination,string dateOfDeparture);
        List<Course> FindAll();
    }
}

File: Bachelor/Semester6-Finals/MPP - Projecting and Programing Methods/tReservation-Lab1/src/main/java/repository/ICourseRepository.java
package repository;

import java.util.List;

public interface ICourseRepository<Integer,Course> {
    int size();
    void save(Course entity);
    void delete(int id);
    void update(int id, Course entity);
    Course findOne(String departureCity, String destination, String dateOfDeparture);
    Course findOne(int courseID);
    List<Course> findAll();
}